### PR TITLE
Basic user management

### PR DESCRIPTION
**Pull request by kenton**
Fix for issues #35 (user management)

### CODE REVIEW

File: app/models/spree/user_decorator.rb [hunk 1]
```diff
@@ -8,6 +8,14 @@ def supplier?
     supplier.present?
   end
 
+  def supplier_admin?
+    supplier? && has_admin_role?
+  end
+
+  def market_maker?
```
Comment (pmn4) [inline app/models/spree/user_decorator.rb:15]: to be clear, a user cannot be both a `market_maker` and a `supplier_admin`?
(totally fine if not, just trying to be clear)

File: app/models/spree/user_decorator.rb [hunk 2]
```diff
@@ -8,6 +8,14 @@ def supplier?
     supplier.present?
   end
 
+  def supplier_admin?
```
Comment (pmn4) [inline app/models/spree/user_decorator.rb:11]: should this method accept an argument (an instance of supplier)? i.e., should a user be a supplier_admin _for_ a particular supplier?
oh wait... a user belongs to a single supplier... never mind.

File: spec/models/spree/user_decorator_spec.rb [hunk 3]
```diff
@@ -19,8 +20,48 @@
...
+      user.spree_roles << admin_role
+
+      expect(user.supplier_admin?).to eq true
+    end
+
+    it "returns false if user is a supplier with no admin role" do
+      expect(user.supplier_admin?).to eq false
+    end
+
+    it "returns false if user is not a supplier but has admin role" do
```
Comment (pmn4) [inline spec/models/spree/user_decorator_spec.rb]: missing `user.spree_roles << admin_role`

File: spec/models/spree/user_decorator_spec.rb [hunk 4]
```diff
@@ -19,8 +20,48 @@
...
+  end
+
+  describe '#market_maker?' do
+    it "returns true if user doesn't have a supplier and has admin role" do
+      user.spree_roles << admin_role
+
+      expect(user.market_maker?).to eq true
+    end
+
+    it "returns false if user is a supplier with admin role" do
```
Comment (pmn4) [inline spec/models/spree/user_decorator_spec.rb]: I don't think the test matches the description

**Comment by kenton** [comment]
@pmn4 all good points.  fixed anything needing fixing.

File: spec/models/spree/user_decorator_spec.rb [hunk 5]
```diff
@@ -19,8 +20,49 @@
...
+  end
+
+  describe '#market_maker?' do
+    it "returns true if user doesn't have a supplier and has admin role" do
+      user.spree_roles << admin_role
+
+      expect(user.market_maker?).to eq true
+    end
+
+    it "returns true if user doesn't have a supplier and doesn't have an admin role" do
```
Comment (pmn4) [inline spec/models/spree/user_decorator_spec.rb]: the bodies of this test and the next are the same, but their descriptions are opposite

**Comment by kenton** [comment]
@pmn4 any other comments on this before I merge?

**Pull request merged**